azurerm/internal/services/appservice/helpers/shared_schema.go

```diff
@@ -0,0 +1,1167 @@
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
+		Computed: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"allowed_origins": {
+					Type:     pluginsdk.TypeSet,
+					Required: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+					},
+				},
+
+				"support_credentials": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  false,
+				},
+			},
+		},
+	}
+}
+
+type SourceControl struct {
+	RepoURL           string `tfschema:"repo_url"`
+	Branch            string `tfschema:"branch"`
+	ManualIntegration bool   `tfschema:"manual_integration"`
+	UseMercurial      bool   `tfschema:"use_mercurial"`
+	RollbackEnabled   bool   `tfschema:"rollback_enabled"`
+}
+
+type SiteCredential struct {
+	Username string `tfschema:"name"`
+	Password string `tfschema:"password"`
+}
+
+func SiteCredentialSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Computed: true,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"name": {
+					Type:     pluginsdk.TypeString,
+					Computed: true,
+				},
+
+				"password": {
+					Type:      pluginsdk.TypeString,
+					Computed:  true,
+					Sensitive: true,
+				},
+			},
+		},
+	}
+}
+
+type AuthSettings struct {
+	Enabled                     bool                    `tfschema:"enabled"`
+	AdditionalLoginParameters   map[string]string       `tfschema:"additional_login_params"`
+	AllowedExternalRedirectUrls []string                `tfschema:"allowed_external_redirect_urls"`
+	DefaultProvider             string                  `tfschema:"default_provider"`
+	Issuer                      string                  `tfschema:"issuer"`
+	RuntimeVersion              string                  `tfschema:"runtime_version"`
+	TokenRefreshExtensionHours  float64                 `tfschema:"token_refresh_extension_hours"`
+	TokenStoreEnabled           bool                    `tfschema:"token_store_enabled"`
+	UnauthenticatedClientAction string                  `tfschema:"unauthenticated_client_action"`
+	AzureActiveDirectoryAuth    []AadAuthSettings       `tfschema:"active_directory"`
+	FacebookAuth                []FacebookAuthSettings  `tfschema:"facebook"`
+	GithubAuth                  []GithubAuthSettings    `tfschema:"github"`
+	GoogleAuth                  []GoogleAuthSettings    `tfschema:"google"`
+	MicrosoftAuth               []MicrosoftAuthSettings `tfschema:"microsoft"`
+	TwitterAuth                 []TwitterAuthSettings   `tfschema:"twitter"`
+}
+
+func AuthSettingsSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
+		Computed: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"enabled": {
+					Type:     pluginsdk.TypeBool,
+					Required: true,
+				},
+
+				"additional_login_params": {
```

Comment:
lets not abbreviate this?
```suggestion
				"additional_login_paramaters": {
```
or
```suggestion
				"additional_login_properties": {
```